src/Parsers/Access/ASTCreateUserQuery.h

```diff
@@ -44,9 +44,8 @@ class ASTCreateUserQuery : public IAST, public ASTQueryWithOnCluster
     std::shared_ptr<ASTUserNamesWithHost> names;
     std::optional<String> new_name;
 
-    std::optional<AuthenticationData> auth_data;
-
-    mutable std::optional<String> temporary_password_for_checks;
+    mutable std::optional<AuthenticationData> auth_data;
```

Comment:
Why is it mutable?

Reply:
In the parser, we do not have the Context to extract the default password type from AccessContol. Passing Context or AccessContol to the parser is a bit hard as it requires changing half of them due to their recursive nature, so we extract the temporary password and encode it in Interpreter and in ClientBase.